osbuild/loop.py

```diff
@@ -598,21 +657,18 @@ def loop_for_fd(self,
                     continue
 
             try:
-                lo.set_fd(fd)
+                lo.configure(fd, **kwargs)
+            except BlockingIOError:
+                lo.clear_fd()
```

Comment:
Will this cause any issues if the exception happens before the fd is set?  Is that possible?

Reply:
I excpect `BlockingIOError` to happen only if fd is set. But I haven't found documentation to back this guess. I'm reading the kernel code atm to figure out.

Reply:
If I'm understanding well the, `BlockingIOError` is there because when `set_fd` failed and in the next round of the loop it succeeds, `set_status` might pop an `EBUSY` error. That's according to the commit #3c953ae4. But in the [doc](https://docs.python.org/3/library/fcntl.html), I don't find evidence that `fnct.ioctl` can raise anything but an OSError.

Also, with the new version we don't need to do sequential calls to `set_fd` and `set_status` everything is done in the kernel in the same mutex protected call https://github.com/torvalds/linux/blob/master/drivers/block/loop.c#L993.

So maybe, it's fine to remove it actually. What do you think @achilleas-k ?